android/src/main/java/com/swmansion/gesturehandler/core/Vector.kt

```diff
@@ -0,0 +1,60 @@
+package com.swmansion.gesturehandler.core
+
+import android.view.VelocityTracker
+import com.swmansion.gesturehandler.core.GestureHandler.Companion.DIRECTION_DOWN
+import com.swmansion.gesturehandler.core.GestureHandler.Companion.DIRECTION_LEFT
+import com.swmansion.gesturehandler.core.GestureHandler.Companion.DIRECTION_RIGHT
+import com.swmansion.gesturehandler.core.GestureHandler.Companion.DIRECTION_UP
+import kotlin.math.hypot
+
+class Vector(var x: Double, var y: Double) {
+    private var unitX: Double = x
+    private var unitY: Double = y
+
+    init {
+        val magnitude = hypot(x, y)
+        val isMagnitudeSufficient = magnitude > MINIMAL_MAGNITUDE
+
+        unitX = if (isMagnitudeSufficient) x / magnitude else 0.0
+        unitY = if (isMagnitudeSufficient) y / magnitude else 0.0
+    }
+
+    private fun computeSimilarity(vector: Vector): Double {
+        return unitX * vector.unitX + unitY * vector.unitY
+    }
+
+    fun isSimilar(vector: Vector, threshold: Double): Boolean {
+        return computeSimilarity(vector) > threshold
+    }
+
+    val magnitude
+        get() = hypot(x, y)
```

Comment:
This also *should* work since `x` and `y` are passed in the constructor
```suggestion
    val magnitude = hypot(x, y)
```